pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1189,6 +1192,73 @@ func (c *Cluster) saveMonConfig() error {
 }
 
 func (c *Cluster) persistExpectedMonDaemons() error {
+	if err := c.persistExpectedMonDaemonsInConfigMap(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in ConfigMap")
+	}
+
+	if err := c.persistExpectedMonDaemonsAsEndpointSlice(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in EndpointSlice")
+	}
+	return nil
+}
+
+func (c *Cluster) persistExpectedMonDaemonsAsEndpointSlice() error {
+	monitors := c.ClusterInfo.AllMonitors()
+
+	if len(monitors) == 0 {
+		logger.Debug("no mon addresses found, skipping endpointslice resource creation")
+		return nil
+	}
+
+	monAddresses := make([]string, 0, len(monitors))
+
+	for _, mon := range monitors {
+		ip := strings.Split(mon.Endpoint, ":")[0]
+		monAddresses = append(monAddresses, ip)
+	}
+
+	endpointSlice := &discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      endpointSliceName,
+			Namespace: c.Namespace,
+			Labels: map[string]string{
+				"kubernetes.io/service-name": endpointSliceName,
+			},
+		},
+		Endpoints: []discoveryv1.Endpoint{
+			{
+				Addresses: monAddresses,
+			},
+		},
+		Ports:       nil,
+		AddressType: "IPv4",
+	}
+
+	logger.Infof("creating mon endpointslice %+v", endpointSlice)
```

Comment:
The entire struct is a bit verbose, how about just the addresses?
```suggestion
	logger.Infof("creating mon endpointslice with endpoints %+v", monAddresses)
```

Reply:
ah thats some log leftovers, removing as it is not required.